tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java

```diff
@@ -69,19 +72,30 @@ public Controller(String muleHome, int timeout) {
     this.domainsDir = new File(muleHome + "/domains");
     this.appsDir = new File(muleHome + "/apps/");
     this.libsDir = new File(muleHome + "/lib/user");
+    this.internalRepository = new File(muleHome, "repository");
     this.wrapperConf = Paths.get(muleHome + "/conf/wrapper.conf");
     this.timeout = timeout != 0 ? timeout : DEFAULT_TIMEOUT;
   }
 
   public abstract String getMuleBin();
 
   public void start(String[] args) {
+    checkRepositoryLocationAndUpdateInternalRepoPropertyIfPresent(args);
     int error = runSync("start", args);
     if (error != 0) {
       throw new MuleControllerException("The mule instance couldn't be started");
     }
   }
 
+  protected void checkRepositoryLocationAndUpdateInternalRepoPropertyIfPresent(String... args) {
+    Optional<String> repoVar =
+        Arrays.asList(args).stream().filter(arg -> arg.contains("-M-DmuleRuntimeConfig.maven.repositoryLocation=")).findFirst();
+    if (repoVar.isPresent()) {
+      this.internalRepository = new File(repoVar.get().split("=")[1]);
```

Comment:
This is a variable of the *Controller* class, so it is pretty clear that it belongs to the runtime and not to any particular application. But calling it internal makes you think it will retrieve the value of the repository that lives inside the distribution